source/extensions/filters/network/dubbo_proxy/router/route_matcher.cc

```diff
@@ -137,19 +141,22 @@ MethodRouteEntryImpl::~MethodRouteEntryImpl() {}
 
 RouteConstSharedPtr MethodRouteEntryImpl::matches(const MessageMetadata& metadata,
                                                   uint64_t random_value) const {
-  if (metadata.hasHeaders() && !RouteEntryImplBase::headersMatch(metadata.headers())) {
+  ASSERT(metadata.hasInvocationInfo());
+  const auto& invocation = static_cast<const RpcInvocationImpl&>(metadata.invocation_info());
```

Comment:
remove `static_cast `?

Reply:
It cannot be deleted here because the static_cast is needed to complete the conversion of the base class reference to the derived class reference, which will fail to compile after deletion.

Reply:
Downcasting, Should use `dynamic_cast` and judge whether the result is nullptr after the cast, rather than use `staic_cast` directly

Reply:
done